Source/WebCore/platform/image-decoders/jpegxl/JPEGXLImageDecoder.cpp

```diff
@@ -437,7 +437,82 @@ LCMSProfilePtr JPEGXLImageDecoder::tryDecodeICCColorProfile()
     return LCMSProfilePtr(cmsOpenProfileFromMem(profileData.data(), profileData.size()));
 }
 
-#endif // USE(LCMS)
+void JPEGXLImageDecoder::maybePerformColorSpaceConversion(void* inputBuffer, void* outputBuffer, unsigned numberOfPixels)
+{
+    if (!m_iccTransform)
+        return;
+
+    cmsDoTransform(m_iccTransform.get(), inputBuffer, outputBuffer, numberOfPixels);
+}
+
+#elif USE(CG)
+
+void JPEGXLImageDecoder::clearColorTransform()
+{
+    m_profile = nullptr;
+}
+
+void JPEGXLImageDecoder::prepareColorTransform()
+{
+    if (m_profile || !m_basicInfo || m_basicInfo->num_extra_channels > 1 || m_basicInfo->exponent_bits_per_sample || m_basicInfo->alpha_exponent_bits)
+        return;
+
+    m_profile = tryDecodeICCColorProfile();
+    // TODO(bugs.webkit.org/show_bug.cgi?id=234222): We should try to use encoded color profile if ICC profile is not available.
+}
+
+RetainPtr<CGColorSpaceRef> JPEGXLImageDecoder::tryDecodeICCColorProfile()
+{
+    size_t profileSize;
+    if (JxlDecoderGetICCProfileSize(m_decoder.get(), &s_pixelFormat, JXL_COLOR_PROFILE_TARGET_DATA, &profileSize) != JXL_DEC_SUCCESS)
+        return nullptr;
+
+    Vector<uint8_t> profileData(profileSize);
+    if (JxlDecoderGetColorAsICCProfile(m_decoder.get(), &s_pixelFormat, JXL_COLOR_PROFILE_TARGET_DATA, profileData.data(), profileData.size()) != JXL_DEC_SUCCESS)
+        return nullptr;
+
+    return adoptCF(CGColorSpaceCreateWithICCData(adoptCF(CFDataCreate(kCFAllocatorDefault, profileData.data(), profileSize)).get()));
```

Comment:
Any reason not to create the `CFData` buffer first, and use that as the target for `JxlDecoderGetColorAsICCProfile`, rather than targeting the temporary vector buffer and copying from it?

Reply:
Yeah, I can use CFMutableData instead. Good idea.